public/app/features/variables/query/reducer.ts

```diff
@@ -109,13 +125,24 @@ const metricNamesToVariableValues = (variableRegEx: string, sort: VariableSort,
     }
 
     if (regex) {
-      matches = regex.exec(value);
-      if (!matches) {
+      matches = getAllMatches(value, regex);
+
+      if (_.isEmpty(matches)) {
+        continue;
+      }
+
+      if (matches.groups && matches.groups.value) {
+        value = matches.groups.value;
+      } else if (matches.groups && matches.groups.hasOwnProperty('value')) {
```

Comment:
Do I understand this correctly that we ignore any falsy values here? Couldn't the value be assigned the text group value if that exists?

Reply:
I had considered that -- but as the value of the variable will almost always be used as a key is some other dashboard metric query I felt that it might introduce unwanted behaviour. I'm not set on that though so if you think it would be better we could include. Whatever is most consistent and understandable.

Reply:
I think we already do this in line 117 above, so my thought was to keep that behavior. https://github.com/grafana/grafana/blob/81699f07d3dc3d6706f7ea7e4a2a24a1db06a836/public/app/features/variables/query/reducer.ts#L117